### PR TITLE
Fix/issue 2651 - Fix notice error on WC tax settings page

### LINKED ISSUE #2651: id key is not found in array error, when taxjar automattic taxes enabled
I want to report a bug either of woocommerce or Automattic woocommerce-services, that causes id key not found error, when automatic taxes calculation is enabled and P.S. Query monitor always shows that requests to taxjar are failing with 406 HTTP response:
[woocommerce/includes/admin/settings/views/settings-tax.php:100](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/admin/settings/views/settings-tax.php#L100)
snippet of code there at the moment (it prepares settings array):
```php
	array( 'type' => 'conflict_error' ), // React mount point for embedded banner slotfill.
```
causes an error of id key not found at:
[woocommerce-services/classes/class-wc-connect-taxjar-integration.php:224](https://github.com/Automattic/woocommerce-services/blob/trunk/classes/class-wc-connect-taxjar-integration.php#L216)
snippet of the code there at the moment:
```php
				if ( ! array_key_exists( $tax_setting['id'], $this->expected_options ) ) {
```
Either `isset($tax_setting['id']) &&` has to be added or this key has to be assigned to a variable in a loop with null coalesce and default value. Or that conflict_error should not exist there. I guess the error is in the latter. :smile:
Branches are main ones. :slightly_smiling_face:

### PR DESCRIPTION
**Pull request by iyut**
## Description
Adding a new condition to check if the tax setting array has id or not.

### Related issue(s)
Fix #2651 

### Steps to reproduce & screenshots/GIFs
1. Install WC >= 7.3.
2. Go to WooCommerce >> Settings >> Tax.
3. No notice error displayed.

### Checklist

- [ ] unit tests

- [x] `changelog.txt` entry added
- [x] `readme.txt` entry added

### CODE REVIEW

**Review by dustinparker — COMMENTED** [review]
That fixed the issue! Nice work! Do you know why the e2e tests are not passing?

**Comment by iyut** [comment]
> That fixed the issue! Nice work! Do you know why the e2e tests are not passing?

Yeah, I think it's been like this for a week or two weeks. The [other merged PR](https://github.com/Automattic/woocommerce-services/pull/2649) is also not passed this test.

**Comment by iyut** [comment]
Hi @dustinparker , I have merged the latest trunk branch into this branch, now the E2E tests is working and passed.

**Review by dustinparker — APPROVED** [review]
LGTM!

File: readme.txt [hunk 1]
```diff
@@ -78,6 +78,9 @@ The source code is freely available [in GitHub](https://github.com/Automattic/wo
 
 == Changelog ==
 
+= 2.2.6 - 2023-xx-xx =
+* Fix    - Fix notice error in WooCommerce tax settings page.
+
 = 2.2.5 - 2023-05-23 =
 * Update - Security update.
 * Fix    - Fix notice error in WooCommerce tax settings page.
```
Comment (dustinparker) [inline readme.txt]: @iyut looks like this line should be removed.

**Pull request merged**